h2o-core/src/test/java/water/runner/CheckKeysTask.java

```diff
@@ -0,0 +1,45 @@
+package water.runner;
+
+import water.*;
+import water.util.ArrayUtils;
+
+import java.util.*;
+
+public class CheckKeysTask extends MRTask<CheckKeysTask> {
+
+    Key[] leakedKeys;
```

Comment:
Is there anyl reason you use an array instead of a List (or Set which is used below)? all this code would be much simpler with a List and I dont see any benefit of an array

Reply:
Yes, we do not have means for moving Set around the cluster implemented (serialization/deserialization) in our core. I used Sets for the local computation.